core/src/main/java/com/scalar/db/transaction/consensuscommit/ParallelExecutor.java

```diff
@@ -76,13 +76,13 @@ public void prepare(List<ParallelExecutorTask> tasks, String transactionId)
           stopOnError,
           "preparation",
           transactionId);
-    } catch (CommitConflictException ignored) {
-      // tasks for preparation should not throw CommitConflictException
+    } catch (ValidationConflictException ignored) {
+      // tasks for preparation should not throw ValidationConflictException
```

Comment:
Ah, yes, that's a good question. That's a tricky part...

Though [ParallelExecutorTask](https://github.com/scalar-labs/scalardb/blob/302a803cf63cdfe9406902fdaa06f76d9237ee72/core/src/main/java/com/scalar/db/transaction/consensuscommit/ParallelExecutor.java#L24-L27) throws `ExecutionException` and `ValidationConflictException`, the actual tasks for `prepare`, `commitRecords`, and `rollbackRecords` don't throw `ValidationConflictException`.

For `prepare`:
https://github.com/scalar-labs/scalardb/blob/302a803cf63cdfe9406902fdaa06f76d9237ee72/core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java#L93-L97

For `commitRecords`:
https://github.com/scalar-labs/scalardb/blob/302a803cf63cdfe9406902fdaa06f76d9237ee72/core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java#L137-L141

For `rollbackRecords`:
https://github.com/scalar-labs/scalardb/blob/302a803cf63cdfe9406902fdaa06f76d9237ee72/core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java#L181-L185

They throw only `ExecutionException` actually.

But, the `validate` task throws both `ValidationConflictException` and `ExecutionException`:
https://github.com/scalar-labs/scalardb/blob/302a803cf63cdfe9406902fdaa06f76d9237ee72/core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java#L318-L405

That's why we can ignore `ValidationConflictException` in the `prepare`, `commitRecords`, and `rollbackRecords` tasks since they can't throw `ValidationConflictException`.

Reply:
Thanks for the explanation. Got it.

So, it sounds catching `ValidationConflictException` in those cases is unexpected. How about throwing `IllegalStateException` when the exception is caught in the `catch` block just in case? My small concern is `ValidationConflictException` might be unexpectedly thrown with future changes and we wouldn't notice it. Of course, we can do that in a different PR.

Reply:
As discussed, we can throw `AssertionError` in that case. I'll fix it in that way. Thank you for pointing it out!

Reply:
Fixed in 8b54c546d3da04501ac1c353ad0ff7f10c4f3acc.

@komamitsu Can you please take a look again when you have time? Thanks.

Reply:
💯 Thank you!